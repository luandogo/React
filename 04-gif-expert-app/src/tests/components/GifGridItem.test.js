import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests in <GifGridItem />', () => {

    const title = 'title';
    const url = 'https://localhost:something.jpg';

    const wrapper = shallow( <GifGridItem title={ title } url={ url }/> )

    
    
    test('should be displayed correctly.', () => {
        
        expect( wrapper ).toMatchSnapshot();
        });
    
    test('should have a paragraph with the title.', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
    });

    test('should have the image of the url and alt from props.  ', () => {

        const img = wrapper.find( 'img' );
        // console.log( img.props() )
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);        
    });

    test('should have animate__fadeIn', () => {

        const div = wrapper.find( 'div' );
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn')).toBe( true );
        
    });
    
    
    

    }
 );