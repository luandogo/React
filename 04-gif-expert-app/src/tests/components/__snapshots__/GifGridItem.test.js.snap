// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> should be displayed correctly. 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="https://localhost:something.jpg"
  />
  <p>
     
    title
     
  </p>
</div>
`;

exports[`Tests in <GifGridItem /> should display the component correctly. 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="https://localhost:something.jpg"
  />
  <p>
     
    title
     
  </p>
</div>
`;
